src/tools/debugger/dosdebug.c

```diff
@@ -518,8 +518,12 @@ int main (int argc, char **argv)
         }
         p[MHP_BUFFERSIZE - 1] = '\0';
         q = strpbrk(p, "\r\n");
-        *q = '\0';
-
+        if (q) {
+          *q = '\0';
+        } else {
+          free(p);
+          p = NULL;
```

Comment:
This code looks very suspicious.
You can call `handle_console_input()` with
normal and freed pointer, depending on something
in a user input?
Maybe this code is correct, but then could it
have fewer magic? Maybe not freeing `p` in the
function?